application/src/main/java/com/softserve/lv460/application/specification/SupportedDeviceSpecification.java

```diff
@@ -0,0 +1,76 @@
+package com.softserve.lv460.application.specification;
+
+import com.softserve.lv460.application.dto.supportedDevice.SupportedDeviceFilterRequest;
+import com.softserve.lv460.application.entity.Feature;
+import com.softserve.lv460.application.entity.SupportedDevice;
+import org.springframework.data.jpa.domain.Specification;
+
+import javax.persistence.criteria.*;
+
+public class SupportedDeviceSpecification implements Specification<SupportedDevice> {
+  private SupportedDeviceFilterRequest filter;
+
+  public SupportedDeviceSpecification(SupportedDeviceFilterRequest filter) {
```

Comment:
since you're using Lombok's constructor annotations, use them everywhere, be consistent